### DISCUSSION TITLE
show columns as rows

### DISCUSSION THREAD

**Discussion by Thornzee** [Budibase/budibase#11171] [Help] [answered]
is there a way to format a table to show columns as rows? My data is already pre-filtered to one row and I would like to display the data like this: 
![columnrow](https://github.com/Budibase/budibase/assets/138253412/46d40988-3c43-46b2-9202-0e078fa53be5)

I tried using a form but I found the formatting to not be what I was looking for.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @Thornzee,

There are a few ways you could get a similar layout to the above. I wouldn't recommend using the table component for this as its purely column based. Nor would I recommend the form/ input fields that goes along with it, as these are quite hard to work with and I believe the `Edit custom css` button only targets the parent div, meaning you would have to do this in a hacky way to get the layout you wish.

Now what i would recommend can be seen below; I've used paragraphs + containers to achieve the layout within your image.

1. Add a data provider (Using a hardcoded value for this demo)
<img width="1464" alt="Screenshot 2023-07-10 at 11 53 05" src="https://github.com/Budibase/budibase/assets/126772285/8e16b550-41cc-4e65-9ce2-9b6ac1f00095">
2. Create a container and add 2 paragraph components to it.
3. Click on the container component
4. Configure the container flex direction etc as shown in the image below;
<img width="1132" alt="Screenshot 2023-07-10 at 11 51 08" src="https://github.com/Budibase/budibase/assets/126772285/95dd35fa-da02-4b0c-a6d5-27d1265d8171">
5. Click the style tab on the right hand of your screen.
6. Scroll down to the `Edit Custom CSS` button and click it.
7. Add the below to the custom CSS into this area;
<img width="1468" alt="Screenshot 2023-07-10 at 11 52 15" src="https://github.com/Budibase/budibase/assets/126772285/a7f5cc9c-f8f9-47a5-8cfa-fbfcdd67cf5b">
8. Click save.
9. Duplicate the container (Will duplicate the paragraphs contained within) and change the names/bindings of the paragraphs fields.

**End result**
<img width="1417" alt="Screenshot 2023-07-10 at 12 13 37" src="https://github.com/Budibase/budibase/assets/126772285/dc8ae3d6-3352-4b3d-a3d0-29c94352ca6e">


I hope this helps!

**Reply by Thornzee** [reply]
Perfect, thank you!!!

**Comment by R2bEEaton** [comment]
If your data provider is a RDBMS like SQL Sever and you can convert it into a Custom Query, perhaps UNPIVOT would be useful as shown in this Stack Overflow answer: https://stackoverflow.com/a/18026337. This would work great if you want to still use tables.

To answer your specific question, to make it look like your picture @ConorWebb96's answer will always be more stable because it is less susceptible to style changes in the future. But if you want to risk it, you can use an Embed element to achieve a more specific style. To do this, set up your structure like this:

![image](https://github.com/Budibase/budibase/assets/34921506/510b3071-6768-40bc-833b-0e61f68c327b)

Make sure the Data Provider only returns one row and set the Form Block to the "View" type. You can use the Configure Fields section to change the labels of your fields. Then add the Embed element, adding this code with any changes you want:

```html
<style>
.spectrum-Form-item label {
  width: 50% !important;
  font-weight: bold !important;
}

.spectrum-Form-item > * {
  border-bottom: 1px solid black !important;
  color: black !important;
}

.spectrum-Form-item .spectrum-Textfield input {
  background-color: rgba(0,0,0,0) !important;
  -webkit-text-fill-color: black !important;
}
</style>
```

Hope this helps!

**Reply by Thornzee** [reply]
Thank you so much!

**Comment by Thornzee** [comment]
Thank you both so much! These answers are great. I have implemented the first suggestion and it works great. I will also look into custom queries to unpivot. 

Thanks for your helpful and thoughtful responses.
